moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/javamodel/reflection/JavaClassImpl.java

```diff
@@ -177,9 +177,13 @@ public Collection<JavaField> getDeclaredFields() {
         Field[] fields = PrivilegedAccessHelper.getDeclaredFields(jClass);
 
         for (Field field : fields) {
-            if (!field.trySetAccessible()) {
-                AbstractSessionLog.getLog().log(SessionLog.FINE, SessionLog.MISC, "set_accessible_in",
-                        "field", field.getName(), jClass.getName());
+            try {
+                if (!field.trySetAccessible()) {
+                    AbstractSessionLog.getLog().log(SessionLog.FINE, SessionLog.MISC, "set_accessible_in",
+                            "field", field.getName(), jClass.getName());
+                }
+            } catch (NoSuchMethodError nsme) {
```

Comment:
Just to be curious. In which exact cases this `catch` happens?

Reply:
dalvik/android does not have the `trySetAccessible()` method from SE 9+